src/base/bittorrent/session.h

```diff
@@ -247,15 +250,18 @@ namespace BitTorrent
         void setTempPathEnabled(bool enabled);
         QString torrentTempPath(const TorrentInfo &torrentInfo) const;
 
-        static bool isValidCategoryName(const QString &name);
-        // returns category itself and all top level categories
-        static QStringList expandCategory(const QString &category);
+        // returns category name itself and all names of top level categories
+        static QStringList expandCategoryName(const QString &categoryName);
 
-        QStringList categories() const;
+        const CategoryMap categories() const;
         QString categorySavePath(const QString &categoryName) const;
-        bool addCategory(const QString &name, const QString &savePath = "");
-        bool editCategory(const QString &name, const QString &savePath);
-        bool removeCategory(const QString &name);
+        // Returns existing TorrentCategory
+        TorrentCategory *findCategory(const QString &categoryName) const;
+        // Returns existing TorrentCategory or creates the new one
+        // Returns nullptr if categoryName is incorrect
```

Comment:
// Returns nullptr if categoryName is not present  ?

Reply:
No. `getCategory()` unlike `findCategory()` tries to create new category if it doesn't exist. So it returns `nullptr` if it fails to create new category (ie. when given category name is incorrect).